src/cmap/connection.ts

```diff
@@ -488,12 +511,19 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
       return;
     }
 
-    this[kStream].end(() => {
+    if (this[kStream].destroyed) {
```

Comment:
I'm curious what inspired this change.  Is this a guard against destroying a destroyed stream?  Or did a test failure necessitate this check?

Reply:
It was a bit of both. Particularly because the Node documentation specifies that after a call to `<readable|writable>.destroy`.

> subsequent calls to write() or end() will result in an ERR_STREAM_DESTROYED error

Also, our `FakeSocket` in the unit test file didn't have an `end` method so we weren't testing for this anywhere and once I added the test to check if the code path was possible, it did fail.

Reply:
Since we always set this.destory to true, shouldn't that prevent us from reaching end after the socket has been destoryed?